packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -463,6 +464,84 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     });
   }
 
+  @override
+  Future<void> handlePointerEventPack(List<PointerEventPack> records) {
+    assert(records != null);
+    assert(records.isNotEmpty);
+    // hitTestHistory is an equivalence of _hitTests in [GestureBinding]
+    final Map<int, HitTestResult> hitTestHistory = <int, HitTestResult>{};
+    return TestAsyncUtils.guard<void>(() async {
+      final DateTime startTime = binding.clock.now();
+      for(final PointerEventPack pack in records) {
+        final Duration timeDiff = pack.timeStamp - binding.clock.now().difference(startTime);
+        if (timeDiff.isNegative) {
+          // Flush all past events
+          // maybe trigger a warning
+          for (final PointerEvent event in pack.events) {
+            _handlePointerEvent(event, hitTestHistory);
+          }
+        }
+        else {
+          // TODO(CareF): reconsider the pumping strategy after
+          // https://github.com/flutter/flutter/issues/60739 is fixed
+          await binding.pump();
+          await binding.executeLater(timeDiff, () async {
+            for (final PointerEvent event in pack.events) {
```

Comment:
We might want to record the actual `timeDiff = pack.timeStamp - binding.clock.now().difference(startTime);` here to see how close it's to 0, and see how well we simulate the timing of those packets. Maybe put this logic and the for loop into a helper function so line 480-482 can use it too.

Reply:
I'm thinking collect `timeDiff` in a list and return it. 

And I realize one more thing. The current code kind of promises the first event (which I offset it to default `Duration.zero`) is almost certainly goes to the `timeDiff.isNegative` branch and becomes a negative timeDiff, which is not a good behavior: it should be rounded to zero.